charts/langsmith/values.yaml

```diff
@@ -107,18 +107,28 @@ config:
       shortlived: "1209600"
   # -- Blob storage configuration
   # Optional. Used to store inputs, outputs, and errors in Blob Storage.
-  # We currently support S3, GCS, and Minio as Blob Storage providers.
+  # We currently support S3, GCS, Minio, and Azure as Blob Storage providers.
   blobStorage:
     enabled: false
+    engine: "S3"
     # If you are using langsmith-managed-clickhouse, you may not want inputs to be stored in clickhouse for search.
     # Set this as false to ensure that inputs/outputs/errors are not stored in clickhouse.
     chSearchEnabled: true
+    # Set this to change the threshold for payloads to be stored in blob storage
+    minBlobStorageSizeKb: "20"
     # If you are using workload identity, you may not need to store the S3 credentials in the secrets.
     # Instead, you will need to add the workload identity annotation to the backend and queue service accounts.
     accessKey: ""
     accessKeySecret: ""
     bucketName: ""
     apiURL: "https://s3.us-west-2.amazonaws.com"
+    # The following blob storage configuration values are for Azure and require blobStorage.engine = "Azure"
+    azureStorageAccountName: ""
+    azureStorageAccountKey: ""
+    azureStorageContainerName: ""
+    azureStorageConnectionString: ""
+    azureStorageSASToken: ""
```

Comment:
Do we need to support all these auth modes? E.g wouldn't really ever expect someone to use SAS or connection string (that's like presignrd url right?)

Reply:
I think connection string is useful for the flexibility - it isn't specifically presigned URL, could use account name and key in the string. would prefer to keep that one.
probably fine to remove SAS token and add support only if needed

Reply:
removed

Reply:
if we do wanna support connection string via secret could we just pass these in via connectionString always? Seems like that parameter should also allow you to specify accountKey or SAStoken? Normal   NotTriggerScaleUp  55s   cluster-autoscaler  pod didn't trigger scale-up: 3 node(s) didn't match Pod's node affinity/selector

Reply:
oops sorry meant accountKey rather than connectionString, think connectionString we should keep for sure since seems like its kind of like apiUrl?

Reply:
kind of nice to construct the URL for them with accountKey and secret imo, even if that's just for teams testing out on Azure

Reply:
ah gotcha, what happens if both are specified today? i think fine maybe just add a note that x overrides this if manually specified. i just generally have a bias towards connectionString when possible since it allows you to add url params and stuff (e.g when specifying db uris)

Reply:
account & key has higher precedence. added a comment here & updated the docs to call this out